src/Module/Xrd.php

```diff
@@ -65,13 +65,20 @@ protected function rawContent(array $request = [])
 
 		if (substr($uri, 0, 4) === 'http') {
 			$name = ltrim(basename($uri), '~');
+			$host = parse_url($uri, PHP_URL_HOST);
 		} else {
 			$local = str_replace('acct:', '', $uri);
 			if (substr($local, 0, 2) == '//') {
 				$local = substr($local, 2);
 			}
 
 			$name = substr($local, 0, strpos($local, '@'));
+			$host = ltrim(substr($local, strpos($local, '@')), '@');
```

Comment:
These two lines can be simplified by using `explode('@', $local)` to fetch the two parts and then just assigning the first part to `$name` and the second to `$host`.